File: client/unity/Assets/Scripts/Assembly-CSharp/CasinoService.cs
using System;
using System.IO;

public class CasinoService
{
	protected static CasinoService instance;

	public static CasinoService gI()
	{
		if (instance == null)
		{
			instance = new CasinoService();
		}
		return instance;
	}

	public void send(Message m)
	{
		Session_ME.gI().sendMessage(m);
		m.cleanup();
	}

	public void requestRoomList()
	{
		Message m = new Message((sbyte)6);
		send(m);
	}

	public void requestBoardList(sbyte id)
	{
		Message message = new Message((sbyte)7);
		try
		{
			message.writer().writeByte(id);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void setMaxPlayer(int max)
	{
		Message message = new Message((sbyte)56);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte((sbyte)max);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void joinBoard(sbyte roomID, sbyte boardID, string pass)
	{
		Message message = new Message((sbyte)8);
		try
		{
			message.writer().writeByte(roomID);
			message.writer().writeByte(boardID);
			message.writer().writeUTF(pass);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void joinAnyBoard()
	{
		Message m = new Message((sbyte)28);
		send(m);
	}

	public void requestStrongest(int page)
	{
		Message message = new Message((sbyte)30);
		try
		{
			message.writer().writeByte(page);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void requestFriendList()
	{
		Message m = new Message((sbyte)(-18));
		send(m);
	}

	public void requestRichest(int page)
	{
		Message message = new Message((sbyte)31);
		try
		{
			message.writer().writeByte(page);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void move(sbyte roomID, sbyte boardID, sbyte[] cards)
	{
		Message message = new Message((sbyte)21);
		try
		{
			message.writer().writeByte(roomID);
			message.writer().writeByte(boardID);
			message.writer().writeByte((sbyte)cards.Length);
			for (int i = 0; i < cards.Length; i++)
			{
				message.writer().writeByte(cards[i]);
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void move(sbyte[] cards)
	{
		Message message = new Message((sbyte)21);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte((sbyte)cards.Length);
			for (int i = 0; i < cards.Length; i++)
			{
				message.writer().writeByte(cards[i]);
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void skip()
	{
		Message message = new Message((sbyte)49);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void forceFinish()
	{
		Message message = new Message((sbyte)53);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void moveCo(sbyte x, sbyte y)
	{
		Message message = new Message((sbyte)21);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte(x);
			message.writer().writeByte(y);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void moveAndWin(int winx, int winy, int windx, int windy, int x, int y)
	{
		Message message = new Message((sbyte)22);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte((sbyte)x);
			message.writer().writeByte((sbyte)y);
			message.writer().writeByte((sbyte)winx);
			message.writer().writeByte((sbyte)winy);
			message.writer().writeByte((sbyte)windx);
			message.writer().writeByte((sbyte)windy);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void firePhom(sbyte cardID)
	{
		Message message = new Message((sbyte)21);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte(cardID);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void GetCardPhom()
	{
		Message message = new Message((sbyte)63);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void eatCardPhom(int[] card, sbyte index)
	{
		Message message = new Message((sbyte)64);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			for (int i = 0; i < card.Length; i++)
			{
				message.writer().writeByte((sbyte)card[i]);
			}
			message.writer().writeByte(index);
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void HaPhomPhom(Card[] myCard)
	{
		Message message = new Message((sbyte)65);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			int num = -1;
			for (int i = 0; i < 10; i++)
			{
				if (myCard[i].phom != 0)
				{
					if (myCard[i].phom != num && num != -1)
					{
						message.writer().writeByte(-1);
					}
					num = myCard[i].phom;
					message.writer().writeByte(myCard[i].cardID);
				}
				else if (num != -1)
				{
					message.writer().writeByte(-1);
					num = -1;
				}
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doAddCardPhom(int[] card)
	{
		Message message = new Message((sbyte)67);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			for (int i = 0; i < card.Length; i++)
			{
				if (card[i] != -1)
				{
					message.writer().writeByte((sbyte)card[i]);
				}
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doResetPhomEatPhom(int[] cardID, int cardEat)
	{
		Message message = new Message((sbyte)68);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte((sbyte)cardEat);
			for (int i = 0; i < 5 && cardID[i] != -1; i++)
			{
				message.writer().writeByte((sbyte)cardID[i]);
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void chatToBoard(string text)
	{
		Message message = new Message((sbyte)9);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeUTF(text);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void leaveBoard(sbyte roomID, sbyte boardID)
	{
		Message message = new Message((sbyte)15);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void ready(sbyte roomID, sbyte boardID, bool isReady)
	{
		Message message = new Message((sbyte)16);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeBoolean(isReady);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void setMoney(sbyte roomID, sbyte boardID, int money)
	{
		Message message = new Message((sbyte)19);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeInt(money);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void setPassword(sbyte roomID, sbyte boardID, string pass)
	{
		Message message = new Message((sbyte)18);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeUTF(pass);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void kick(sbyte roomID, sbyte boardID, int kickID)
	{
		Message message = new Message((sbyte)11);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeInt(kickID);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void startGame(sbyte roomID, sbyte boardID)
	{
		Message message = new Message((sbyte)20);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void createCell(Point[][] array)
	{
		int num = 0;
		Message message = new Message((sbyte)64);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			for (int i = 0; i < 8; i++)
			{
				for (int j = 0; j < 8; j++)
				{
					if (array[i][j].isRemove)
					{
						array[i][j].isRemove = false;
						message.writer().writeByte((sbyte)(i * 8 + j));
						num++;
					}
				}
			}
			send(message);
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public void doMoveDiamond(int iSelected, int selected)
	{
		Message message = new Message((sbyte)21);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte((sbyte)iSelected);
			message.writer().writeByte((sbyte)selected);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doSkipDaimond()
	{
		Message message = new Message((sbyte)49);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void doOutPath()
	{
		Message message = new Message((sbyte)24);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
	}

	public void PutMoneyOk(MyVector info, sbyte room, sbyte board)
	{
		Message message = new Message((sbyte)21);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			if (info.size() > 0)
			{
				for (int i = 0; i < info.size(); i++)
				{
					PimgBC pimgBC = (PimgBC)info.elementAt(i);
					message.writer().writeByte((sbyte)pimgBC.moneyPut);
					pimgBC.moneyPut = 0;
				}
			}
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e, "PutMoneyOkzz()");
		}
	}

	public void ta(sbyte room, sbyte board, sbyte indexFrom, sbyte indexTo)
	{
		Message message = new Message((sbyte)65);
		try
		{
			message.writer().writeByte(BoardScr.roomID);
			message.writer().writeByte(BoardScr.boardID);
			message.writer().writeByte(room);
			message.writer().writeByte(board);
			message.writer().writeByte(indexFrom);
			message.writer().writeByte(indexTo);
			send(message);
		}
		catch (IOException e)
		{
			Out.logError(e, "ta()");
		}
	}

	public void skip(sbyte roomID, sbyte boardID)
	{
		Message m = new Message((sbyte)49);
		send(m);
	}
}

File: client/java/src/avt/CasinoService.java
package avt;

import java.io.IOException;
import java.util.Vector;

public final class CasinoService extends IService {
   private static CasinoService instance;

   public static CasinoService gI() {
      if (instance == null) {
         instance = new CasinoService();
      }

      return instance;
   }

   public final void requestRoomList() {
      this.createMessage((byte)6);
      this.sendMessage();
   }

   public final void requestBoardList(byte var1) {
      this.createMessage((byte)7);
      this.writeByte(var1);
      this.sendMessage();
   }

   public final void setMaxPlayer(int var1) {
      try {
         this.createMessageWithBoard((byte)56);
         super.m.writer().writeByte(var1);
      } catch (Exception var3) {
      }

      this.sendMessage();
   }

   public final void joinBoard(byte var1, byte var2, String var3) {
      this.createMessage((byte)8);

      try {
         super.m.writer().writeByte(var1);
         super.m.writer().writeByte(var2);
         super.m.writer().writeUTF(var3);
      } catch (IOException var5) {
      }

      this.sendMessage();
   }

   public final void joinAnyBoard() {
      this.createMessage((byte)28);
      this.sendMessage();
   }

   public final void requestFriendList() {
      this.createMessage((byte)-18);
      this.sendMessage();
   }

   public final void move(byte[] var1) {
      try {
         this.createMessageWithBoard((byte)21);
         super.m.writer().writeByte(var1.length);

         for(int var2 = 0; var2 < var1.length; ++var2) {
            super.m.writer().write(var1[var2]);
         }
      } catch (Exception var3) {
      }

      this.sendMessage();
   }

   public final void skipTurn() {
      try {
         this.createMessageWithBoard((byte)49);
      } catch (Exception var2) {
      }

      this.sendMessage();
   }

   public final void haBai() {
      try {
         this.createMessageWithBoard((byte)21);
      } catch (Exception var2) {
      }

      this.sendMessage();
   }

   public final void createMessageWithBoard(byte var1) throws IOException {
      super.createMessage(var1);
      super.m.writer().writeByte(BoardScr.roomID);
      super.m.writer().writeByte(BoardScr.boardID);
   }

   public final void moveCo(byte var1) {
      try {
         this.createMessageWithBoard((byte)21);
         super.m.writer().writeByte(var1);
      } catch (Exception var3) {
      }

      this.sendMessage();
   }

   public final void GetCardPhom() {
      try {
         this.createMessageWithBoard((byte)63);
      } catch (Exception var2) {
      }

      this.sendMessage();
   }

   public final void eatCardPhom(int[] var1, byte var2) {
      try {
         this.createMessageWithBoard((byte)64);

         for(int var3 = 0; var3 < var1.length; ++var3) {
            super.m.writer().writeByte(var1[var3]);
         }

         super.m.writer().writeByte(var2);
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void HaPhomPhom(Card[] var1) {
      try {
         this.createMessageWithBoard((byte)65);
         byte var2 = -1;

         for(int var3 = 0; var3 < 10; ++var3) {
            if (var1[var3].phom != 0) {
               if (var1[var3].phom != var2 && var2 != -1) {
                  super.m.writer().writeByte(-1);
               }

               var2 = var1[var3].phom;
               super.m.writer().writeByte(var1[var3].cardID);
            } else if (var2 != -1) {
               super.m.writer().writeByte(-1);
               var2 = -1;
            }
         }
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void doAddCardPhom(int[] var1) {
      try {
         this.createMessageWithBoard((byte)67);

         for(int var2 = 0; var2 < var1.length; ++var2) {
            if (var1[var2] != -1) {
               super.m.writer().writeByte(var1[var2]);
            }
         }
      } catch (Exception var3) {
      }

      this.sendMessage();
   }

   public final void doResetPhomEatPhom(int[] var1, int var2) {
      try {
         this.createMessageWithBoard((byte)68);
         super.m.writer().writeByte(var2);

         for(var2 = 0; var2 < 5 && var1[var2] != -1; ++var2) {
            super.m.writer().writeByte(var1[var2]);
         }
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void chatToBoard(String var1) {
      try {
         this.createMessageWithBoard((byte)9);
         super.m.writer().writeUTF(var1);
      } catch (IOException var3) {
      }

      this.sendMessage();
   }

   public final void leaveBoard() {
      try {
         this.createMessageWithBoard((byte)15);
      } catch (IOException var2) {
      }

      this.sendMessage();
   }

   public final void ready(boolean var1) {
      try {
         this.createMessageWithBoard((byte)16);
         super.m.writer().writeBoolean(var1);
      } catch (IOException var3) {
      }

      this.sendMessage();
   }

   public final void setMoney(int var1) {
      try {
         this.createMessageWithBoard((byte)19);
         super.m.writer().writeInt(var1);
      } catch (IOException var3) {
      }

      this.sendMessage();
   }

   public final void setPassword(String var1) {
      try {
         this.createMessageWithBoard((byte)18);
         super.m.writer().writeUTF(var1);
      } catch (IOException var3) {
      }

      this.sendMessage();
   }

   public final void kick(int var1) {
      try {
         this.createMessageWithBoard((byte)11);
         super.m.writer().writeInt(var1);
      } catch (IOException var3) {
      }

      this.sendMessage();
   }

   public final void startGame() {
      try {
         this.createMessageWithBoard((byte)20);
      } catch (IOException var2) {
      }

      this.sendMessage();
   }

   public final void createCell(Point[][] var1) {
      try {
         this.createMessageWithBoard((byte)64);

         for(int var2 = 0; var2 < 8; ++var2) {
            for(int var3 = 0; var3 < 8; ++var3) {
               if (var1[var2][var3].isRemove) {
                  var1[var2][var3].isRemove = false;
                  super.m.writer().writeByte((var2 << 3) + var3);
               }
            }
         }
      } catch (Exception var4) {
         var4.printStackTrace();
      }

      this.sendMessage();
   }

   public final void doMoveDiamond(int var1, int var2) {
      try {
         this.createMessageWithBoard((byte)21);
         super.m.writer().writeByte(var1);
         super.m.writer().writeByte(var2);
      } catch (IOException var4) {
         var4.printStackTrace();
      }

      this.sendMessage();
   }

   public final void doOutPath() {
      try {
         this.createMessageWithBoard((byte)24);
      } catch (IOException var2) {
         var2.printStackTrace();
      }

      this.sendMessage();
   }

   public final void PutMoneyOk(Vector var1) {
      try {
         this.createMessageWithBoard((byte)21);
         if (var1.size() > 0) {
            for(int var2 = 0; var2 < var1.size(); ++var2) {
               PimgBC var3 = (PimgBC)var1.elementAt(var2);
               super.m.writer().writeByte(var3.moneyPut);
               var3.moneyPut = 0;
            }
         }
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void ta(byte var1, byte var2) {
      try {
         this.createMessageWithBoard((byte)65);
         super.m.writer().writeByte(var1);
         super.m.writer().writeByte(var2);
      } catch (Exception var4) {
      }

      this.sendMessage();
   }

   public final void skip() {
      try {
         this.createMessageWithBoard((byte)49);
      } catch (Exception var2) {
      }

      this.sendMessage();
   }
}
